### DISCUSSION TITLE
Multi-Step Ahead Forecasting

### DISCUSSION THREAD

**Discussion by ghost** [pycaret/pycaret#1197] [Q&A]
Does pycaret support multi-step ahead forecasting for time series data like stock market?

**Comment by AdamTheDestroyer** [comment]
Hi @veioxve3r1d23seb2g6b 
everything depends on your data. 

In each "How-to" in the great World Wide Web, they describe it like this:

You always separate Data you know for the future and Data you want to know for the future. 
Data you know for the future are always Time Data like: 
Amount of WorkingDays in a week, Holidays, WeekNumber, Quarter, Year, Month, Weather forecast Data and maybe something like special events you know in the future(Football events and other sport events).

You train your Model with all the "Time Data" as features ("Time Data" has to be past till today and not including future Data!)  and your "Target Data".
Example: "Time Data" and "Target Data" from 01.01.2016 until 21.09.2021 (today).

After finalizing your model, you predict your target (you want to know for the future) with future "Time Data" as input (as described above).
Example: "Time Data" from 21.09.2021 until 31.12.2021 (future).

That's how i role the dices ;-)

I hope one time there will be a possibility to do automatic multitarget predictions. For Example: If you have excellent Data that describes your "Target Data", you always have to do the above described procedure for every feature to generate future values. In the second step, you use all the future values of your features to predict your real target. In my case ~ 3000 features for 6 Groups my prediction takes  35 hours. (3000 targets, so 3000 times generating individual "Future Data" to predict with all the 3000 generated "Future Data" 1 time the real target )

**Comment by ngupta23** [comment]
@veioxve3r1d23seb2g6b - There is a time series module in development, but you can try out the alpha release using the following.

```
> pip install pyraret-ts-alpha
```

Please make sure that this is installed in a separate environment from pycaret for now.

Here are some starting notebooks that show how to build models with this: **https://github.com/pycaret/sktime-dev-days-2021**
